lib/active_model/serializer/adapter/attributes.rb

```diff
@@ -46,10 +62,59 @@ def fragment_cache(cached_hash, non_cached_hash)
 
         private
 
+        def relevant_included_associations_for(serializer, nested_associations = {})
```

Comment:
This is a smell for a latent object that you're passing serializer in when you already have it as a reader